app/helpers/application_helper.rb

```diff
@@ -37,6 +37,12 @@ def user_signing_up?
     !current_user || !current_user.two_factor_enabled?
   end
 
+  def session_with_trust?
+    current_user ||
+      current_page?(controller: 'sign_up/email_confirmations', action: 'create') ||
```

Comment:
maybe put the `current_page?` checks in their own method? I'm guessing the exclusion list will grow over time. `current_page_requires_trust?`